flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperatorFactory.java

```diff
@@ -58,11 +56,11 @@ public void setMailboxExecutor(MailboxExecutor mailboxExecutor) {
 	}
 
 	@Override
-	public StreamOperator createStreamOperator(StreamTask containingTask, StreamConfig config, Output output) {
+	public <T extends StreamOperator<OUT>> T createStreamOperator(StreamOperatorInitializer<OUT> initializer) {
```

Comment:
👍 to the idea. Long overdue. However, `StreamOperatorInitializer` sounds like something active, while it's just a parameter object. How about `StreamOperatorSettings` or `StreamOperatorParameters`?

Reply:
+1, other candidates: `StreamOperatorCreationContext`, `StreamOperatorFactoryContext`